Adding support for gcloud scope aliases

Temikus opened the pull request:
Fixing #36. Adding gcloud compute aliases support to be consistent with official utility.

ihmccreery commented on lib/fog/google/models/compute/server.rb:
```diff
@@ -206,6 +206,32 @@ def add_ssh_key username, key
+            'storage-rw' => 'devstorage.read_write'
+              }
+
+          scope_array.map! {
```
This is mutating `scope_array`, (`map!` is destructive).  I'd either switch to non-destructive `map` and returning a new array, (I think this is what I'd prefer,) or entitle the method `map_service_accounts!`, and probably call it in the constructor.  Mutating class variables during `save` sounds scary to me.

Stylistically, for a multi-line block, you should use the `do end` syntax, and use brackets for inline.  See [this answer](http://stackoverflow.com/a/5587403/594530).

ihmccreery commented on lib/fog/google/models/compute/server.rb:
```diff
@@ -206,6 +206,32 @@ def add_ssh_key username, key
         end
 
+        def map_service_accounts(scope_array)
+          gce_scope_aliases = {
```
`gce_scope_aliases` seems like it should be a constant, declared in the class scope or somewhere else.

ihmccreery commented on lib/fog/google/models/compute/server.rb:
```diff
@@ -206,6 +206,32 @@ def add_ssh_key username, key
+
+          scope_array.map! {
+            |e| gce_scope_aliases[e] ? gce_scope_aliases[e] : e
+              }
```
Make sure your blocks are indented and outdented properly and consistently.

ihmccreery commented:
Hey Artem, thanks for this work!  See inline comments.

Temikus commented:
Hi Ike,

Consistently outdented blocs (used previous indentation model), moved the aliases away into a constant. Squashed the commits.
Let me know if I missed something.

ihmccreery commented:
Hi Artem,

You're still doing a destructive operation during `save`:

> This is mutating `scope_array`, (`map!` is destructive). I'd either switch to non-destructive `map` and returning a new array, (I think this is what I'd prefer,) or entitle the `method map_service_accounts!`, and probably call it in the constructor. Mutating class variables during `save` sounds scary to me.

Unless you have a strong argument in favor of doing so, I recommend doing either a non-destructive operation, or switching to call this method during initialization.

Also,

> Stylistically, for a multi-line block, you should use the do end syntax, and use brackets for inline. See this [answer](http://stackoverflow.com/questions/5587264/do-end-vs-curly-braces-for-blocks-in-ruby/5587403#5587403).

Temikus commented:
Hi Ike,

Thanks for the comments!

1) About destructive `map!`
Sorry, missed that particular one, my bad. Agreed, changed to a non-destructive map.

2) About one line if's
Initially, I expanded that statement only because it was over 80 chars. Rubocop indicated that to me, however, you already had a multiline statement like this:
```
              "scopes" => service_accounts.map {
                |w| w.start_with?("https://") ? w : "https://www.googleapis.com/auth/#{w}"
              }
```
, and the unless/if block looks a little messier so wasn't sure how to proceed.
I now mapped it in a full if/else block.

Branch updated, commits squashed. Let me know if you have any comments.

Cheers,
A.

ihmccreery commented:
Looks good to me!  I don't really have a preference for `_ ? _ : _` versus `if`/`else` syntax, but this looks fine.

Thanks for all your work on this!
